### PR TITLE
A CONTRIBUTING.md file

### PR DESCRIPTION
**Pull request by mortberg**
As discussed in https://github.com/agda/cubical/issues/317.

I wrote down everything I could think of but there's surely things I forgot and things that should be reformulated. So comments are very welcome!

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+CONTRIBUTING
+============
+
+We very much appreciate contributions in the form of pull requests (PRs).
+When preparing a PR here are some general guidelines:
+
+- Please read through and clean your code before making a PR. Clean
+  code has reasonable line length (<100 characters), good indentation,
```
Comment (mortberg) [inline CONTRIBUTING.md:8]: I know that 80 +- epsilon is also a common limit for line lengths. I have no really strong opinions about this as long as it's less than 100

**Comment by mortberg** [comment]
Maybe we should say something about structuring files with `Foo/Base.agda`, `Foo/Properties.agda` and then having `Foo.agda` exporting these in some form or another. Wasn't there a discussion started by @ecavallo about what `Foo.agda` should export at some point?

**Comment by mortberg** [comment]
Maybe we should also say something about implicit arguments. I like to have as many things implicit as possible, especially universe levels, but one should not overdo it and make arguments implicit that Agda can't infer.

**Comment by mortberg** [comment]
One more thing: it's not good to have files that take a long time to typecheck. If typechecking a file starts becoming slow it is often a sign that some definitions need to be optimized (maybe using copattern-matching or abstract or rethinking that maths).

**Comment by ayberkt** [comment]
@mortberg It can perhaps be mentioned that `isSetΠn` and `isPropΠn` are to be preferred over multiple applications of their unary counterparts. Same for `funExt`.

This can save some time for the maintainers. I, for instance, was not aware of the existence of the n-ary versions initially.

**Comment by mortberg** [comment]
PRs with postulates should be marked as Draft PRs until the postulates have been fixed

**Comment by mortberg** [comment]
I'm reasonably happy with this now. Any comments before I merge?

**Comment by mortberg** [comment]
Alright, merging this now. Anyone should feel free to make updates

**Pull request merged**